DoSomething/gambit-conversations
# Chore/224 add request id

# What does this PR do?
It injects the `request_id` into most logs that have access to the `req` or `res` objects, which contain the `request_id` metadata.

This solution doesn't add the `request_id` to **ALL** logs when processing a single request. This solution is a good compromise to update the logs already in place in very crucial spots, like middleware and helpers with access to `req` and `res` objects.

We can likely increase the usefulness of our logs if we continue to add them in all middleware in all routes and injecting the `request_id` on them.

# How to test?
- 👀 
- On staging, you can signup and reportback. You should see the `request_id` in many generated logs.
- Using Kibana, you can isolate a `request_id` and see a clear history of the steps taken to get a response. [Here is an example](http://logs.dosomething.org/app/kibana#/discover?_g=(refreshInterval:('$$hashKey':'object:2085',display:'10%20seconds',pause:!f,section:1,value:10000),time:(from:now-7d,mode:quick,to:now))&_a=(columns:!(message,conversationId,campaignId,text,signupId,userId,messageTemplate),filters:!(('$$hashKey':'object:790','$state':(store:appState),meta:(alias:!n,disabled:!f,index:'filebeat-*',key:heroku_drain_id,negate:!f,value:d.1a141963-f3ea-42b0-b6bc-4a219b34adba),query:(match:(heroku_drain_id:(query:d.1a141963-f3ea-42b0-b6bc-4a219b34adba,type:phrase)))),('$$hashKey':'object:1254771','$state':(store:appState),meta:(alias:!n,disabled:!f,index:'filebeat-*',key:request_id,negate:!f,value:f7d3eaa8-16a9-4969-8b93-ddb0d980e599),query:(match:(request_id:(query:f7d3eaa8-16a9-4969-8b93-ddb0d980e599,type:phrase))))),index:'filebeat-*',interval:auto,query:(query_string:(analyze_wildcard:!t,query:'*')),sort:!('@timestamp',desc),vis:(aggs:!((params:(field:request_id,orderBy:'2',size:20),schema:segment,type:terms),(id:'2',schema:metric,type:count)),type:histogram))&indexPattern=filebeat-*&type=histogram).

# Relevant issues
Fixes #224
